src/new-components/button-link/button-link.stories.mdx

```diff
@@ -478,6 +478,92 @@ export function LoadingButtonLink(props) {
     return <ButtonLink {...props} loading={loading} onClick={(event) => setLoading(true)} />
 }
 
+### Full-width
+
+export function FullWidthTemplate({ label, variant, tone, size }) {
+    if (label === 'Click me <em>now</em>') {
+        label = (
+            <>
+                Click me <em>now</em>
+            </>
+        )
+    }
+    const props = {
+        variant,
+        tone,
+        size,
+        width: 'full',
+        startIcon: <Icon />,
+        endIcon: <Icon />,
+    }
+    return (
+        <Stack space="medium">
+            <Heading level="2">Full-width buttons and label alignment</Heading>
+            <Text>
+                When buttons have <code>width="full"</code> they can also customize how the label is
+                aligned horizontally.
+            </Text>
+            <Box width="small">
+                <Stack space="medium">
+                    <LoadingButtonLink {...props} align="left">
+                        {label}
+                    </LoadingButtonLink>
+                    <LoadingButtonLink {...props} align="center">
+                        {label}
+                    </LoadingButtonLink>
+                    <LoadingButtonLink {...props} align="right">
+                        {label}
+                    </LoadingButtonLink>
+                </Stack>
+            </Box>
+        </Stack>
+    )
+}
+
+<Canvas>
+    <Story
+        parameters={{ docs: { source: { type: 'code' } } }}
+        argTypes={{
+            label: {
+                control: { type: 'select' },
+                options: [
+                    'OK',
+                    'Submit',
+                    'Mark as done',
+                    'Yes, cancel my subscription',
+                    'Click me <em>now</em>',
+                    'If you click me now, youʼll take away the biggest part of me',
+                ],
+                defaultValue: 'Submit',
+            },
+            variant: {
+                options: ['primary', 'secondary', 'tertiary', 'quaternary'],
+                control: { type: 'select' },
+                defaultValue: 'primary',
+            },
+            tone: {
+                options: ['normal', 'destructive'],
+                control: { type: 'inline-radio' },
+                defaultValue: 'normal',
+            },
+            size: {
+                options: ['small', 'normal', 'large'],
+                control: { type: 'inline-radio' },
+                defaultValue: 'normal',
+            },
+            disabled: { control: false },
+            startIcon: { control: false },
+            endIcon: { control: false },
+            icon: { control: false },
+            tooltip: { control: false },
+            exceptionallySetClassName: { control: false },
+        }}
+        name="Full-width"
+    >
+        {FullWidthTemplate.bind({})}
```

Comment:
I guess we can only pick between having controls vs having code snippets. Right now this shows up like this when you click on show code:

![image](https://user-images.githubusercontent.com/8531248/163137757-16387687-9443-4883-8579-986840ca9f16.png)

Reply:
Yup, something we need to figure out with storybook. Perhaps it has bothered someone else already and there's a way around it. Not an issue introduced here, as all the other examples using templates are the same.